Ginger/GingerCoreNET/Logger/WebReportGenerator.cs

```diff
@@ -129,61 +130,48 @@ public void DeleteFoldersData(string clientAppFolderPath)
 
         private void PopulateMissingFields(LiteDbRunSet liteDbRunSet, string clientAppPath)
         {
-            string imageFolderPath = Path.Combine(clientAppPath, "assets", "screenshots");
-            List<string> runSetEnv = new List<string>();
-            int totalRunners = liteDbRunSet.RunnersColl.Count;
-            int totalPassed = 0;
-            if (liteDbRunSet.RunStatus == eRunStatus.Automated.ToString())
-            {
-                totalPassed = liteDbRunSet.RunnersColl[0].BusinessFlowsColl.Where(bf => bf.RunStatus == eRunStatus.Passed.ToString()).Count();
-                liteDbRunSet.RunnersColl[0].RunStatus = liteDbRunSet.RunnersColl[0].BusinessFlowsColl[0].RunStatus;
-            }
-            else
+            //select template 
+            HTMLReportConfiguration _HTMLReportConfig = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>().Where(x => (x.IsDefault == true)).FirstOrDefault();
+
+            //populate data based on level
+            if (string.IsNullOrEmpty(_HTMLReportConfig.ExecutionCalculationLevel))
             {
-                totalPassed = liteDbRunSet.RunnersColl.Where(runner => runner.RunStatus == eRunStatus.Passed.ToString()).Count();
+                _HTMLReportConfig.ExecutionCalculationLevel = HTMLReportConfiguration.ReportsExecutionCalculationLevel.Activity.ToString();
             }
-            int totalExecuted = totalRunners - liteDbRunSet.RunnersColl.Where(runner => runner.RunStatus == eRunStatus.Pending.ToString() || runner.RunStatus == eRunStatus.Skipped.ToString() || runner.RunStatus == eRunStatus.Blocked.ToString()).Count();
-            if (totalRunners != 0)
-                liteDbRunSet.ExecutionRate = string.Format("{0:F1}", (totalExecuted * 100 / totalRunners).ToString());
-            if (totalRunners != 0)
-                liteDbRunSet.PassRate = string.Format("{0:F1}", (totalPassed * 100 / totalRunners).ToString());
-            liteDbRunSet.Elapsed = liteDbRunSet.RunnersColl.Sum(a => a.Elapsed);
 
+            string imageFolderPath = Path.Combine(clientAppPath, "assets", "screenshots");
+            List<string> runSetEnv = new List<string>();
+
+            if (liteDbRunSet.ChildExecutableItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] != 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4549121)